api/src/opentrons/protocol_reader/protocol_reader.py

```diff
@@ -41,24 +42,26 @@ def __init__(
             file_identifier: File identifier. Default impl. used if None.
             role_analyzer: File role analyzer. Default impl. used if None.
             file_format_validator: File format validator. Default impl. used if None.
+            file_hasher: File hasher. Default impl. used if None.
         """
         self._file_reader_writer = file_reader_writer or FileReaderWriter()
         self._file_identifier = file_identifier or FileIdentifier()
         self._role_analyzer = role_analyzer or RoleAnalyzer()
         self._file_format_validator = file_format_validator or FileFormatValidator()
+        self._file_hasher = file_hasher or FileHasher()
 
-    async def read_and_save(
-        self, files: Sequence[AbstractInputFile], directory: Path
+    async def save(
+        self, files: Sequence[BufferedFile], directory: Path, content_hash: str
     ) -> ProtocolSource:
-        """Compute a `ProtocolSource` from file-like objects and save them as files.
+        """Compute a `ProtocolSource` from buffered files and save them as files.
 
         The input is parsed and statically analyzed to ensure it's basically
         well-formed. For example, labware definition files must conform to the labware
         definition schema.
 
         Arguments:
-            files: List of files-like objects. Do not attempt to reuse any objects
-                objects in this list once they've been passed to the ProtocolReader.
+            files: List buffered files. Do not attempt to reuse any objects
+                in this list once they've been passed to the ProtocolReader.
```

Comment:
```suggestion
            files: List of buffered files
```

Since you've changed the interface to accept `BufferedFile`s instead of `AbstractInputFile`s, we don't need this quirky warning about file reuse anymore. This had to do with file `.seek()`ing, but we don't need to worry about that with `BufferedFile`s.